reporter-config3/src/main/java/com/addthis/metrics3/reporter/config/InfluxDBReporterConfig.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.addthis.metrics3.reporter.config;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.addthis.metrics.reporter.config.AbstractInfluxDBReporterConfig;
+import com.addthis.metrics.reporter.config.HostPort;
+import com.codahale.metrics.MetricRegistry;
+import com.izettle.metrics.influxdb.InfluxDbHttpSender;
+import com.izettle.metrics.influxdb.InfluxDbReporter;
+import com.izettle.metrics.influxdb.InfluxDbSender;
+
+public class InfluxDBReporterConfig extends AbstractInfluxDBReporterConfig implements MetricsReporterConfigThree
+{
+    private static final Logger log = LoggerFactory.getLogger(InfluxDBReporterConfig.class);
+
+    private InfluxDbReporter reporter;
+
+    private void enableMetrics3(HostPort hostPort, MetricRegistry registry) throws Exception
+    {
+        InfluxDbSender influxDbSender = new InfluxDbHttpSender(getProtocol(), hostPort.getHost(), hostPort.getPort(),
+            getDbName(), getAuth(), getRealRateunit(), getConnectionTimeout(), getReadTimeout());
+
+        reporter = InfluxDbReporter.forRegistry(registry).convertRatesTo(getRealRateunit())
+            .convertDurationsTo(getRealDurationunit()).withTags(getResolvedTags())
+            .filter(MetricFilterTransformer.generateFilter(getPredicate())).build(influxDbSender);
+
+        reporter.start(getPeriod(), getRealTimeunit());
+    }
+
+    @Override
+    public void report()
+    {
+        if (reporter != null) {
+            reporter.report();
+        }
+    }
+
+    @Override
+    public boolean enable(final MetricRegistry registry)
+    {
+        boolean success = checkClass("com.izettle.metrics.influxdb.InfluxDbReporter");
+        if (!success)
+        {
+            return false;
+        }
+
+        List<HostPort> hosts = getFullHostList();
+        if (hosts == null || hosts.isEmpty())
+        {
+            log.error("No hosts specified, cannot enable InfluxDBReporter");
+            return false;
+        }
+
+        for (HostPort hostPort : hosts)
+        {
+            log.info("Enabling InfluxDBReporter to {}:{}", hostPort.getHost(), hostPort.getPort());
+            try
+            {
+                // connect with first working host
```

Comment:
Sorry, I am looking at the influx docs and it seems clustering have changed in latest releases. I was still thinking about old built in clustering, which is now not available. You are right, LB and InfluxDB Relay are probably now the expected solution, so I will change this to require 1 host.

btw I have looked into other reporters before but I have not found consistency:
1. [statsd](https://github.com/addthis/metrics-reporter-config/blob/master/reporter-config3/src/main/java/com/addthis/metrics3/reporter/config/StatsDReporterConfig.java#L33) creates reporter for every host and reports to all of them 
2. [graphite](https://github.com/addthis/metrics-reporter-config/blob/master/reporter-config3/src/main/java/com/addthis/metrics3/reporter/config/GraphiteReporterConfig.java#L69)  creates reporter for  every configured host but actually reports only to the [last one](https://github.com/addthis/metrics-reporter-config/blob/master/reporter-config3/src/main/java/com/addthis/metrics3/reporter/config/GraphiteReporterConfig.java#L51) (because each of them reassignes the `reporter` reference)?